File: Editorial/0171_Excel_Sheet_Column_Number/CSharp/1_Right_to_Left.cs
public class Solution {
    public int TitleToNumber(string s) {
        int result = 0;

        Dictionary<Char, int> alpha_map = new Dictionary<Char, int>();
        for (int i = 0; i < 26; i++) {
            // Decimal 65 in ASCII corresponds to char 'A'
            Char c = Convert.ToChar(i + 65);
            alpha_map[c] = i + 1;
        }

        int n = s.Length;
        for (int i = 0; i < n; i++) {
            char cur_char = s[n - 1 - i];
            result += (alpha_map[cur_char] * ((int)Math.Pow(26, i)));
        }

        return result;
    }
}

File: Editorial/0101-0200/0171_Excel_Sheet_Column_Number/TypeScript/1_Right_to_Left.ts
function titleToNumber(s: string): number {
    let result: number = 0;

    const alpha_map: { [key: string]: number } = {};
    for (let i: number = 0; i < 26; i++) {
        // Decimal 65 in ASCII corresponds to char 'A'
        alpha_map[String.fromCharCode(i + 65)] = i + 1;
    }

    const n: number = s.length;
    for (let i: number = 0; i < n; i++) {
        const cur_char: string = s[n - 1 - i];
        result += alpha_map[cur_char] * Math.pow(26, i);
    }
    return result;
}
